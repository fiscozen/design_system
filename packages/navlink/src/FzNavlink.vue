<script lang="ts" setup generic="T">
import { FzIcon } from '@fiscozen/icons'
import { FzNavlinkProps } from './types'
import { commonClasses } from './classUtils'

defineProps<FzNavlinkProps<T>>()
</script>

<template>
  <button
    :disabled="disabled"
    :class="[
      { 'flex w-32 flex-row items-center justify-center': iconOnly, 'px-12 py-6': !iconOnly },
      commonClasses
    ]"
  >
    <FzIcon
      :name="iconName"
      v-if="iconName"
      :class="iconOnly ? '' : 'mr-8'"
      :size="iconSize"
    ></FzIcon>
    <slot>
      <span>{{ label }}</span>
    </slot>
  </button>
</template>
