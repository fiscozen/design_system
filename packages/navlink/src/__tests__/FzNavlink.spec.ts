import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import FzNavlink from '../FzNavlink.vue'

describe('FzNavlink', () => {
  it('should render correctly with label', () => {
    const wrapper = mount(FzNavlink, {
      props: {
        label: 'this is a test label',
        iconName: 'bell'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
    wrapper.setProps({ disabled: true })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render correctly if icon only', () => {
    const wrapper = mount(FzNavlink, {
      props: {
        iconName: 'bell'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
    wrapper.setProps({ disabled: true })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('span').text()).to.be.empty
  })

  it('should emit native event', () => {
    const wrapper = mount(FzNavlink, {
      props: {
        label: 'this is a test label'
      }
    })
    wrapper.trigger('click')

    const clickEvs = wrapper.emitted('click')
    expect(clickEvs).toHaveLength(1)
  })
  it('should not emit click when disabled', () => {
    const wrapper = mount(FzNavlink, {
      props: {
        label: 'this is a test label',
        disabled: true
      }
    })
    wrapper.trigger('click')

    const clickEvs = wrapper.emitted('click')
    expect(clickEvs).toBeUndefined()
  })
})
