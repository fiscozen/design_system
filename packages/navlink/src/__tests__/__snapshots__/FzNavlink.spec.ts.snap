// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FzNavlink > should render correctly if icon only 1`] = `
"<button class="font-medium flex w-32 flex-row items-center justify-center text-grey-500 hover:bg-background-alice-blue border-1 focus:border-1 focus:bg-background-alice-blue disabled:text-grey-100 disabled:bg-core-white h-32 items-center rounded border-transparent text-sm hover:text-blue-500 focus:border-blue-500 focus:text-blue-500 disabled:border-transparent hover:font-medium focus:font-medium"><svg class="svg-inline--fa fa-bell fa-lg" aria-hidden="true" focusable="false" data-prefix="fasl" data-icon="bell" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512">
    <path class="" fill="currentColor" d="M208 0h32V32.8c80.9 8 144 76.2 144 159.2v97.4l59.3 59.3 4.7 4.7V360v40 16H432 16 0V400 360v-6.6l4.7-4.7L64 289.4V192c0-83 63.1-151.2 144-159.2V0zm16 64C153.3 64 96 121.3 96 192V296v6.6l-4.7 4.7L32 366.6V384H416V366.6l-59.3-59.3-4.7-4.7V296 192c0-70.7-57.3-128-128-128zM160 448h32c0 17.7 14.3 32 32 32s32-14.3 32-32h32c0 35.3-28.7 64-64 64s-64-28.7-64-64z"></path>
  </svg><span></span></button>"
`;

exports[`FzNavlink > should render correctly if icon only 2`] = `
"<button class="font-medium flex w-32 flex-row items-center justify-center text-grey-500 hover:bg-background-alice-blue border-1 focus:border-1 focus:bg-background-alice-blue disabled:text-grey-100 disabled:bg-core-white h-32 items-center rounded border-transparent text-sm hover:text-blue-500 focus:border-blue-500 focus:text-blue-500 disabled:border-transparent hover:font-medium focus:font-medium"><svg class="svg-inline--fa fa-bell fa-lg" aria-hidden="true" focusable="false" data-prefix="fasl" data-icon="bell" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512">
    <path class="" fill="currentColor" d="M208 0h32V32.8c80.9 8 144 76.2 144 159.2v97.4l59.3 59.3 4.7 4.7V360v40 16H432 16 0V400 360v-6.6l4.7-4.7L64 289.4V192c0-83 63.1-151.2 144-159.2V0zm16 64C153.3 64 96 121.3 96 192V296v6.6l-4.7 4.7L32 366.6V384H416V366.6l-59.3-59.3-4.7-4.7V296 192c0-70.7-57.3-128-128-128zM160 448h32c0 17.7 14.3 32 32 32s32-14.3 32-32h32c0 35.3-28.7 64-64 64s-64-28.7-64-64z"></path>
  </svg><span></span></button>"
`;

exports[`FzNavlink > should render correctly with label 1`] = `
"<button class="font-medium px-12 py-6 text-grey-500 hover:bg-background-alice-blue border-1 focus:border-1 focus:bg-background-alice-blue disabled:text-grey-100 disabled:bg-core-white h-32 items-center rounded border-transparent text-sm hover:text-blue-500 focus:border-blue-500 focus:text-blue-500 disabled:border-transparent hover:font-medium focus:font-medium"><svg class="svg-inline--fa fa-bell fa-lg mr-8" aria-hidden="true" focusable="false" data-prefix="fasl" data-icon="bell" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512">
    <path class="" fill="currentColor" d="M208 0h32V32.8c80.9 8 144 76.2 144 159.2v97.4l59.3 59.3 4.7 4.7V360v40 16H432 16 0V400 360v-6.6l4.7-4.7L64 289.4V192c0-83 63.1-151.2 144-159.2V0zm16 64C153.3 64 96 121.3 96 192V296v6.6l-4.7 4.7L32 366.6V384H416V366.6l-59.3-59.3-4.7-4.7V296 192c0-70.7-57.3-128-128-128zM160 448h32c0 17.7 14.3 32 32 32s32-14.3 32-32h32c0 35.3-28.7 64-64 64s-64-28.7-64-64z"></path>
  </svg><span>this is a test label</span></button>"
`;

exports[`FzNavlink > should render correctly with label 2`] = `
"<button class="font-medium px-12 py-6 text-grey-500 hover:bg-background-alice-blue border-1 focus:border-1 focus:bg-background-alice-blue disabled:text-grey-100 disabled:bg-core-white h-32 items-center rounded border-transparent text-sm hover:text-blue-500 focus:border-blue-500 focus:text-blue-500 disabled:border-transparent hover:font-medium focus:font-medium"><svg class="svg-inline--fa fa-bell fa-lg mr-8" aria-hidden="true" focusable="false" data-prefix="fasl" data-icon="bell" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512">
    <path class="" fill="currentColor" d="M208 0h32V32.8c80.9 8 144 76.2 144 159.2v97.4l59.3 59.3 4.7 4.7V360v40 16H432 16 0V400 360v-6.6l4.7-4.7L64 289.4V192c0-83 63.1-151.2 144-159.2V0zm16 64C153.3 64 96 121.3 96 192V296v6.6l-4.7 4.7L32 366.6V384H416V366.6l-59.3-59.3-4.7-4.7V296 192c0-70.7-57.3-128-128-128zM160 448h32c0 17.7 14.3 32 32 32s32-14.3 32-32h32c0 35.3-28.7 64-64 64s-64-28.7-64-64z"></path>
  </svg><span>this is a test label</span></button>"
`;
